source/security.txt

```diff
@@ -15,6 +26,411 @@ Secure Your Data
    /security/enterprise-authentication
    /security/in-use-encryption
 
-- :ref:`pymongo-auth`
-- :ref:`pymongo-enterprise-auth`
-- :ref:`pymongo-in-use-encryption`
\ No newline at end of file
+Overview
+--------
+
+MongoDB supports multiple mechanisms that you can use to authenticate your application.
+This page contains code examples that show each of these mechanisms.
+
+.. tip::
+  
+  To learn more about any of the authentication mechanisms on this page, see the
+  :ref:`pymongo-auth` and :ref:`pymongo-enterprise-auth` pages.
+
+To use an authentication example from this page, copy the code example into the
+:ref:`sample application <pymongo-auth-sample>` or your own application.
+Be sure to replace all placeholders in the code examples, such as ``<hostname>``, with
+the relevant values for your MongoDB deployment.
+
+.. _pymongo-auth-sample:
+
+Sample Application
+~~~~~~~~~~~~~~~~~~
+
+You can use the following sample application to experiment with authenticating to
+MongoDB. Replace the highlighted lines with an authentication code example from this
+page.
```

Comment:
S: I think a little more information would be helpful here. I also wonder if it's better to use the comments to indicate where to paste code, rather than the highlight in case the highlight doesn't show or is hard to see. Maybe something like:
```suggestion
You can use the following sample application to experiment with authenticating to
MongoDB. To use the sample application, create a new python application with {+driver-short+} installed, and paste in the sample application code. Then copy an authentication code example from this page and paste it between the comments labeled ``authentication code here``.
```

Reply:
reworded